src/main/java/seedu/address/model/task/DeadlineTask.java

```diff
@@ -3,28 +3,50 @@
 import java.time.LocalDateTime;
 import java.util.Objects;
 
+import java.util.function.Predicate;
+
 import seedu.address.commons.exceptions.IllegalValueException;
 
 public class DeadlineTask extends Task {
 
-    private static final String FMT_STRING = "DeadlineTask[name=%s, due=%s]";
+    private static final String FMT_STRING = "DeadlineTask[name=%s, due=%s, finished=%s]";
 
     private final LocalDateTime due;
 
-    public DeadlineTask(Name name, LocalDateTime due) {
+    private boolean finished;
+
+    public static Predicate<? super DeadlineTask> isNotFinishedDeadline() {
+        return p -> !p.isFinished();
+    }
+
+    public DeadlineTask(Name name, LocalDateTime due, boolean finished) {
         super(name);
         assert due != null;
         this.due = due;
+        this.finished = finished;
+    }
+
+    public DeadlineTask(Name name, LocalDateTime due) {
+        this(name, due, false);
     }
 
     public DeadlineTask(String name, LocalDateTime due) throws IllegalValueException {
-        this(new Name(name), due);
+        this(new Name(name), due, false);
     }
 
     public LocalDateTime getDue() {
         return due;
     }
 
+    public boolean isFinished() {
+        return this.finished;
+    }
+
+    public void markAsFinished() {
+        assert this.finished == false;
+        this.finished = true;
+    }
```

Comment:
To add on: it will also make @potatowagon 's life harder because we won't be able to just do a shallow copy anymore to save the state the taskbook for undo/redo (The task instance will keep its previous values)